DOCS-2541: Move component APIs (#3504)
viamrobotics/docs: docs/components/arm/_index.md

# SME: Peter L
---

> Arms provide an API for linear motion planning, including self-collision prevention and obstacle avoidance.

Reviewer: ```suggestion
The arm component provides an API for linear motion planning, including self-collision prevention and obstacle avoidance.
```
I'd word it this way so we know we're talking about the Viam component instead of the concept of the hardware itself.

Also what does "linear" mean here? Should that come out?

Author: Clarifying information about components is really beyond the scope of the PR. I can do that where we know the answer but this is an already enormous PR, and getting into fixing content when this is really just reorganizing things, is beyond what I am willing to do here. 

I am guessing it is linear because you do not define waypoints, it is strictly from the current point to a new defined point.

Reviewer: Okay. Finished reviewing with this in mind.
I think there are waypoints along the path though, and it's definitely not a straight line if there are obstacles in the way, and also, the end effector often sweeps in curves because the geometry of an arm often makes this more efficient. I'll follow up myself about the "linear" thing to check accuracy.

If you have a physical robotic arm, consisting of a serial chain of joints and links, with a fixed end and an end effector end, use an arm component.

> Typically, arms have two ends: one fixed in place, and one with a device you can position.

Reviewer: ```suggestion
Arms have two ends: one fixed in place, and one with a device you can position.
```
what other kind of arm is there? Even with a delta arm there's sort of a base and definitely an end effector.

> When operating an arm you can place its end effector at arbitrary cartesian positions relative to the base of the arm.

Reviewer: ```suggestion
When controlling an arm, you can place its end effector at arbitrary cartesian positions relative to the base of the arm.
```
"controlling" seems to me like a more correct word here because it gets more to the point of the API, whereas in my head "operating" somehow implies driving it with a joystick or similar.

> When operating an arm you can place the its end effector at arbitrary cartesian positions relative to the base of the arm.

Reviewer 2: ```suggestion
When operating an arm you can place its end effector at arbitrary cartesian positions relative to the base of the arm.
```

Each arm model is supported with a driver that is compatible with the software API that the model's manufacturer supports.
While some arm models build inverse kinematics into their software, many do not.

- Most of the arm drivers for the Viam RDK bypass any onboard inverse kinematics, and use Viam's [motion service](/services/motion/) instead.

- This driver handles turning the arm on and off, querying the arm for its current joint position, sending requests for the arm to move to a specified set of joint positions, and engaging brakes as needed, if supported.

Arm drivers are also paired, in the RDK, with JSON files that describe the kinematics parameters of each arm.

- When you configure a supported arm model to connect to `viam-server`, the Arm driver will load and parse the kinematics file for the Viam RDK's [frame system](/services/frame-system/) service to use.

- The [frame system](/services/frame-system/) will allow you to easily calculate where any part of your machine is relative to any other part, other machine, or piece of the environment.

- All arms have a `Home` position, which corresponds to setting all joint angles to 0.

> - When an arm is moved with a `move_to_position` call, the movement will follow a straight line, and not deviate from the start or end orientations more than the start and orientations differ from one another

Reviewer 2: ```suggestion
- When an arm is moved with a `move_to_position` call, the movement will follow a straight line, and not deviate from the start or end orientations more than the start and orientations differ from one another.
```
